### ISSUE TITLE
Openxc-Dump 'wa' Error

### ISSUE THREAD

**Issue by ahassa** [openxc/openxc-python#91]
Dump throws error when trying to write trace file to local value. Changing file open mode 'wa' to 'w' in base seems to fix it

**Comment by emarsman** [comment]
https://github.com/openxc/openxc-python/blob/7b22cac7e5f311c871aa339de6d06b4f774a63f2/openxc/sources/base.py#L127

**Comment by peplin** [comment]
What's the error you are getting? Adding `a` means "append to the file if it exists instead of overwriting", so changing from `wa` to only `w` changes the behavior.

**Comment by ahassa** [comment]
Don’t remember the exact error off the top of my head, other than the Dump command gave open file error with ‘wa’ being unknown or unrecognized.

**Comment by peplin** [comment]
Was it something like this:

```
>>> open('test', 'wa')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: must have exactly one of create/read/write/append mode
```

if so, you may be running Python 3 - unfortunately the library only works with Python 2.x.

**Comment by jwschmo** [comment]
Wa wa

**Comment by ahassa** [comment]
Something along those lines, though I was using Python 2.12

**Comment by emarsman** [comment]
I can't find the documentation now on when, but I think this did change in a recent python 2.7 release. According to (https://docs.python.org/2.7/library/functions.html#open):

> 'w' for writing (truncating the file if it already exists), and 'a' for appending 

w truncates. a appends. Those are conflicting if both are used. 

> emarsman@WGC1W74RD9XF2  ~/Documents/GitHub
> $ python --version
> Python 2.7.12
> emarsman@WGC1W74RD9XF2  ~/Documents/GitHub
> $ openxc-dump --log-mode file
> Traceback (most recent call last):
>   File "C:\Python27\Scripts\openxc-dump-script.py", line 9, in <module>
>     load_entry_point('openxc==0.13.0', 'console_scripts', 'openxc-dump')()
>   File "c:\python27\lib\site-packages\openxc\tools\dump.py", line 35, in main
>     source = source_class(callback=receive, **source_kwargs)
>   File "c:\python27\lib\site-packages\openxc\sources\usb.py", line 49, in __init__
>     super(UsbDataSource, self).__init__(**kwargs)
>   File "c:\python27\lib\site-packages\openxc\sources\base.py", line 174, in __init__
>     super(BytestreamDataSource, self).__init__(**kwargs)
>   File "c:\python27\lib\site-packages\openxc\sources\base.py", line 44, in __init__
>     self.logger = SourceLogger(self, log_mode)
>   File "c:\python27\lib\site-packages\openxc\sources\base.py", line 127, in __init__
>     self.file = open(filename, 'wa')
> ValueError: Invalid mode ('wa')
> 


Since the files create are timestamped upon creation, this should just be 'w'.

**Comment by emarsman** [comment]
fixed in 0.14.0-dev+
